src/components/dialogs/custom-mui-dialog.js

```diff
@@ -0,0 +1,105 @@
+import { FormProvider } from 'react-hook-form';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import { Grid } from '@mui/material';
+import { FormattedMessage } from 'react-intl';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import makeStyles from '@mui/styles/makeStyles';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        margin: 0,
+        padding: theme.spacing(2),
+    },
+    closeButton: {
+        position: 'absolute',
+        right: theme.spacing(1),
+        top: theme.spacing(1),
+        color: theme.palette.grey[500],
+    },
+    dialogPaper: {
+        width: 'auto',
+        minWidth: '800px',
+        margin: 'auto',
+    },
+    content: {
+        overflow: 'auto',
+        justifyContent: 'space-around',
+        flexGrow: 1,
+    },
+}));
+
+const CustomMuiDialog = ({
+    name,
+    open,
+    schema,
+    methods,
+    onClose,
+    onSave,
+    onValidationError,
+    titleId,
+    disabledSave,
+    removeOptional,
+    ...dialogProps
+}) => {
+    const classes = useStyles();
+    const { handleSubmit } = methods;
+
+    const handleClose = (event) => {
+        onClose(event);
+    };
+
+    const handleCancel = (event) => {
+        onClose(event);
+    };
+
+    const handleValidate = (data) => {
+        onSave(data);
+        onClose(data);
+    };
+
+    const handleValidationError = (errors) => {
+        onValidationError && onValidationError(errors);
+    };
+    return (
+        <FormProvider
+            validationSchema={schema}
+            {...methods}
+            removeOptional={removeOptional}
+        >
+            <Dialog
+                classes={{ paper: classes.dialogPaper }}
+                open={open}
+                onClose={handleClose}
+                fullWidth
+            >
+                <DialogTitle>
+                    <Grid container>
+                        <Grid item xs={11}>
+                            <FormattedMessage id={titleId} />
+                        </Grid>
+                    </Grid>
+                </DialogTitle>
+                <DialogContent>{dialogProps.children}</DialogContent>
+                <DialogActions>
+                    <Button onClick={handleCancel}>
+                        <FormattedMessage id="cancel" />
+                    </Button>
+                    <Button
+                        disabled={disabledSave}
```

Comment:
Use isDirty : https://github.com/gridsuite/gridstudy-app/blob/fe947e88ac5f254166b62481aa19692cebe6c84e/src/components/dialogs/commons/submitButton.js#L12

Reply:
This whole file is a copy of an old implementation of modification-dialog. We should put that in commons-ui and use the same implementation instead of duplicating old code and having them diverge between gridstudy and gridexplore.

Reply:
That's the plan, we just decided to do the RHF and AGGrid migration by copying progressively the file we need, and then choose what and how to migrate the RHF components into commons ui.
But yeah we should use the latest versions and try to do not modify them too much.

Reply:
done